source/connect/csot.txt

```diff
@@ -1,210 +1,218 @@
-one or more operations can execute before control is returned to the app. This
-timeout applies to all of the work done to execute the operation, including
-server selection, connection checkout, serialization, and
-server-side execution.
+Overview
+--------
 
-Basic Usage
------------
+With {+driver-short+}, you can specify a **client side operation timeout**, a time limit
+for server operations to run. The timeout applies all steps needed to complete the
+operation, including server selection, connection checkout, serialization, and server-side
+execution. When the timeout expires, {+driver-short+} raises a timeout exception.
 
-The following example uses the ``~pymongo.timeout()`` method to configure a 10-second
-timeout for a ``~pymongo.collection.Collection.insert_one()``.
+You can specify a timeout in two ways: by using the ``timeout()`` method or by using
+the ``timeoutMS`` connection option.
+
+timeout() Method
+----------------
+
+To specify a timeout in your code, call the ``timeout()`` method and pass the timeout
+length, in seconds, as an argument. You must call the ``timeout()`` method
+in a ``with`` statement, as shown in the following example:
 
 .. code-block:: python
 
-   import pymongo
    with pymongo.timeout(10):
-       coll.insert_one({"name": "Nunu"})
+       collection.insert_one({"name": "Yngwie"})
 
-.. note::
-  
-   The ``~pymongo.timeout()`` method applies to all {+driver-short+} operations inside the
-   ``with`` block.
+In the preceding example, if the insert operation doesn't finish within 10 seconds,
+{+driver-short+}, raises an error.
 
-The following example ensures that both the ``insert_one()`` and ``find()`` methods
-finish within 10 seconds total. If they don't, {+driver-short+} raises a timeout error.
+The timeout you specify applies to all operations inside the ``with`` block. In the
+following example, the insert and find operations must both finish within a total of
+10 seconds:
 
 .. code-block:: python
 
    with pymongo.timeout(10):
-       coll.insert_one({"name": "Nunu"})
-       coll.find_one({"name": "Nunu"})
+       coll.insert_one({"name": "Yngwie"})
+       coll.find_one({"name": "Yngwie"})
 
-If you nest calls to the ``~pymongo.timeout()`` method, the deadline of the inner method
-call is capped by the deadline of the outer method call. The deadline can only be
-shortened, not extended.
+Nested Timeout Blocks
+~~~~~~~~~~~~~~~~~~~~~
+
+When execution enters a nested ``timeout()`` block, the outer ``timeout()`` deadline is
+paused. When execution exits the nested ``timeout()`` block, the previous ``timeout()``
+deadline resumes.
 
-When execution enters a nested ``timeout`` block, the outer ``timeout`` deadline is paused.
-When execution exits the nested ``timeout`` block, the previous ``timeout`` deadline resumes.
 The following example shows how nested calls to the ``timeout()`` method interact:
 
 .. code-block:: python
 
    with pymongo.timeout(5):
```

Comment:
It might be helpful to do a quick walkthrough of this after the code example. Especially to explain that 10 second timeout that's not being used. Thinking back to our code comment conversation the other day, if for some reason a user can't read/understand the comments in that code block then this would probably still be unclear